ma_gym/envs/lumberjacks/lumberjacks.py

```diff
@@ -0,0 +1,351 @@
+
+        # Convert it from numpy array
+        obs = obs.tolist()
+
+        if self.full_observable:
+            obs = [feature for agent_obs in obs for feature in agent_obs]
+            obs = [obs] * self.n_agents
+
+        return obs
+
+    def _agent_generator(self) -> Tuple[int, Agent]:
+        """Yields agent_id and agent for all agents in environment."""
+        for agent_id, agent in enumerate(self._agents):
+            yield agent_id, agent
+
+    def _agent_view_generator(self, pos: Coordinates, view_range: Tuple[int, int]):
+        """Yields position, number of agent and tree strength for cells in distance of `view_range` from `pos`.  """
+        mask = (
+            slice(pos[0] - view_range[0], pos[0] + view_range[0] + 1),
+            slice(pos[1] - view_range[1], pos[1] + view_range[1] + 1),
+        )
+        yield from self._view_generator(mask)
+
+    def step(self, agents_action: List[int]):
+        assert len(agents_action) == self.n_agents
+
+        self._step_count += 1
+        rewards = np.full(self.n_agents, self._step_cost)
+
+        # Move agents
+        for (agent_id, agent), action in zip(self._agent_generator(), agents_action):
+            if not self._agent_dones[agent_id]:
+                self._update_agent_pos(agent, action)
+
+        # Cut down trees
+        mask = (np.sum(self._agent_map, axis=2) >= self._tree_map) & (self._tree_map > 0)
+        self._tree_map[mask] = 0
+
+        # Calculate rewards
+        rewards += np.sum(mask * self._tree_cutdown_reward, axis=(0, 1))
+        self._total_episode_reward += rewards
+
+        if (self._step_count >= self._max_steps) or (np.count_nonzero(self._tree_map) == 0):
+            self._agent_dones = [True] * self.n_agents
+
+        return self.get_agent_obs(), rewards.tolist(), self._agent_dones, {}
+
+    def _update_agent_pos(self, agent: Agent, move: int):
+        """Moves `agent` according the `move` command."""
+        next_pos = self._next_pos(agent.pos, move)
+
+        # Remove agent from old position
+        self._agent_map[agent.pos[0], agent.pos[1], agent.id] = 0
+
+        # Add agent to the new position
+        agent.pos = next_pos
+        self._agent_map[next_pos[0], next_pos[1], agent.id] = 1
+
+    def _next_pos(self, curr_pos: Coordinates, move: int) -> Coordinates:
+        """Returns next valid position in extended coordinates given by `move` command relative to `curr_pos`."""
+        if move == ACTIONS_IDS['noop']:
+            next_pos = curr_pos
+        elif move == ACTIONS_IDS['down']:
+            next_pos = (curr_pos[0] + 1, curr_pos[1])
+        elif move == ACTIONS_IDS['left']:
+            next_pos = (curr_pos[0], curr_pos[1] - 1)
+        elif move == ACTIONS_IDS['up']:
+            next_pos = (curr_pos[0] - 1, curr_pos[1])
+        elif move == ACTIONS_IDS['right']:
+            next_pos = (curr_pos[0], curr_pos[1] + 1)
+        else:
+            raise ValueError(f'Unknown action {move}.')
```

Comment:
It will be nice to show set of valid actions in the error message. 

Eg:
Unknown action {move}.  Valid action should be in range 0 to 5.